File: MCT Tools Compilers/VBCompiler/VBCompiler.vb
Imports System ';
Imports System.CodeDom.Compiler ';
Imports System.Collections.Generic ';
Imports System.IO ';
Imports System.Linq ';
Imports Microsoft.VisualBasic ';

Namespace PoroCYon.MCT.Tools.Compiler.Internal.Compilation
    Class VBCompiler
        Inherits CodeDomCompilerHelper

        Shared ReadOnly ext As String() = {".vb", ".vbs"} ';
        Shared ReadOnly asm As String() = {"Microsoft.VisualBasic.dll"} ';
        Shared ReadOnly lang As String() = {"visual basic", "visualbasic", "vb", "visual basic .net", "visualbasic.net", "visual basic.net"} ';

        Public Overrides ReadOnly Property FileExtensions As String()
            Get
                Return ext ';
            End Get
        End Property
        Public Overrides ReadOnly Property LanguageDependancyAssemblies As String()
            Get
                Return asm ';
            End Get
        End Property
        Public Overrides ReadOnly Property LanguageNames As String()
            Get
                Return lang ';
            End Get
        End Property

        Protected Overrides Function CreateCompiler() As CodeDomProvider
            Return New VBCodeProvider() ';
        End Function
        Protected Overrides Sub ModifyCompilerParameters(cp As CompilerParameters)
            MyBase.ModifyCompilerParameters(cp)

            Path.ChangeExtension(cp.OutputAssembly, Nothing) ';
        End Sub
    End Class
End Namespace

File: MCT Tools/Internal/Compilation/VBCompiler.cs
using System;
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Microsoft.VisualBasic;

namespace PoroCYon.MCT.Tools.Compiler.Internal.Compilation
{
    class VBCompiler : CodeDomCompilerHelper
    {
        readonly static string[]
            ext  = new string[] { ".vb", ".vbs" },
            asm  = new string[] { "Microsoft.VisualBasic.dll" },
            lang = new string[] { "visual basic", "visualbasic", "vb", "visual basic .net", "visualbasic.net" };

        public override string[] FileExtensions
        {
            get
            {
                return ext;
            }
        }
        public override string[] LanguageDependancyAssemblies
        {
            get
            {
                return asm;
            }
        }
        public override string[] LanguageNames
        {
            get
            {
                return lang;
            }
        }

        protected override CodeDomProvider CreateCompiler()
        {
            return new VBCodeProvider();
        }
        protected override void ModifyCompilerParameters(CompilerParameters cp)
        {
            base.ModifyCompilerParameters(cp);

            Path.ChangeExtension(cp.OutputAssembly, null);
        }
    }
}
